### ISSUE TITLE
setup3.py should have support for ConditionalExtension

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8462]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8462|
|Type|enhancement|
|Created|2016-06-13 09:34:25Z|
|Branch|https://github.com/twisted/twisted/tree/8462-rodrigc-setup|

setup3.py should have support for ConditionalExtension so that
we can test installing of C modules.
Currently setup3.py skips installing of all C modules

<details><summary>Searchable metadata</summary>

```
trac-id__8462 8462
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8462_rodrigc_setup 8462-rodrigc-setup
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1465810465378576 1465810465378576
changetime__1467017721405299 1467017721405299
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/182

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc set owner to @hawkowl|
|-|-|

Please look.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

Sent to buildbot

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc removed owner|
|-|-|

Patch moved to:
https://github.com/twisted/twisted/pull/227

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_6"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Looks good to me, feel free to merge.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|GitHub set status to `closed`|
|-|-|

In changeset e39465c6ac04cb130913b74895b55b33ab2183d4
```
#!CommitTicketReference repository="" revision="e39465c6ac04cb130913b74895b55b33ab2183d4"
Merge 8462-rodrigc-setup: Add support in setup3.py for installing extensions

Author: rodrigc
Reviewer: Lukasa
Fixes: #8462
```
